aspnetcore/fundamentals/file-providers.md

```diff
@@ -1,148 +1,261 @@
 
-[!code-html[](file-providers/sample/src/FileProviderSample/Views/Home/Index.cshtml?highlight=2,7,9,11,15)]
+Inject the provider into any class constructor and assign it to a local field. Use the field throughout the class's methods to access files.
 
-The result:
+::: moniker range=">= aspnetcore-2.0"
 
-![File provider sample application listing physical files and folders](file-providers/_static/physical-directory-listing.png)
+In the sample app, the `IndexModel` class receives an `IFileProvider` instance to obtain directory contents for the app's base path.
 
-### EmbeddedFileProvider
+*Pages/Index.cshtml.cs*:
+
+[!code-csharp[](file-providers/samples/2.x/FileProviderSample/Pages/Index.cshtml.cs?name=snippet1)]
+
+The `IDirectoryContents` are iterated in the page.
+
+*Pages/Index.cshtml*:
+
+[!code-cshtml[](file-providers/samples/2.x/FileProviderSample/Pages/Index.cshtml?name=snippet1)]
+
+::: moniker-end
+
+::: moniker range="< aspnetcore-2.0"
+
+In the sample app, the `HomeController` class receives an `IFileProvider` instance to obtain directory contents for the app's base path.
+
+*Controllers/HomeController.cs*:
+
+[!code-csharp[](file-providers/samples/1.x/FileProviderSample/Controllers/HomeController.cs?name=snippet1)]
+
+The `IDirectoryContents` are iterated in the view.
+
+*Views/Home/Index.cshtml*:
 
-The `EmbeddedFileProvider` is used to access files embedded in assemblies. In .NET Core, you embed files in an assembly with the `<EmbeddedResource>` element in the *.csproj* file:
+[!code-cshtml[](file-providers/samples/1.x/FileProviderSample/Views/Home/Index.cshtml?name=snippet1)]
 
-[!code-json[](file-providers/sample/src/FileProviderSample/FileProviderSample.csproj?range=13-18)]
+::: moniker-end
 
-You can use [globbing patterns](#globbing-patterns) when specifying files to embed in the assembly. These patterns can be used to match one or more files.
+::: moniker range=">= aspnetcore-2.0"
+
+### ManifestEmbeddedFileProvider
+
+The [ManifestEmbeddedFileProvider](/dotnet/api/microsoft.extensions.fileproviders.manifestembeddedfileprovider) is used to access files embedded within assemblies. The `ManifestEmbeddedFileProvider` uses a manifest compiled into the assembly to reconstruct the original paths of the embedded files.
 
 > [!NOTE]
-> It's unlikely you would ever want to actually embed every .js file in your project in its assembly; the above sample is for demo purposes only.
+> The `ManifestEmbeddedFileProvider` is available in ASP.NET Core 2.1 or later. To access files embedded in assemblies in ASP.NET Core 2.0 or earlier, see the [ASP.NET Core 1.x version of this topic](xref:fundamentals/file-providers?view=aspnetcore-1.1).
+
+To generate a manifest of the embedded files, set the **&lt;GenerateEmbeddedFilesManifest&gt;** property to `true`. Specify the files to embed with [&lt;EmbeddedResource&gt;](/dotnet/core/tools/csproj#default-compilation-includes-in-net-core-projects):
```

Comment:
I've been code-fencing XML element names.

Reply:
> I've been code-fencing XML element names.

Since XML is a markup language and the guidelines call for markup to be bold, I went this way (and have done this in other spots). If we're bolding for HTML (per guidelines), imo it's logical to bold for XML.

@Rick-Anderson @tdykstra What interpretation have you taken?

Reply:
I agree with Scott on code format for XML element names.  Based on our contrib guide guidelines I would code style HTML elements as well.

https://review.docs.microsoft.com/en-us/help/contribute/text-formatting-guidelines?branch=master

Reply:
I can't see what you posted. I have it in two spots ...

Section 12.3 Formatting developer text elements
Section 20.4 Formatting text in instructions

| Element | Convention | Example
| --- | --- | --- |
| Markup language elements (tags) | Bold. Capitalization varies. | **\<img>**<br>**\<input type=text>**<br>**\<!DOCTYPE html>** |

Did they change it, or do they have conflicting guidelines?

Reply:
Yes in a sense conflicting but that guide does include a disclaimer:  "Your group's editorial style guide might contain additional conventions."